File: Techschool/Techschool/ClientApp/src/app/models/vacations/annual-vacation-form.model.ts
import { AnnualVacationModel } from './annual-vacation.model';
import { VacationFormModel } from './vacation-form.model';

export class AnnualVacationFormModel extends VacationFormModel {
  startOfWorkingYear: Date;
  endOfWorkingYear: Date;
  days: number;
  annualVacations: AnnualVacationModel[];
}

File: Techschool/Techschool.BLL/Models/Vacations/AnnualVacationFormModel.cs
using System;
using System.Collections.Generic;

namespace Techschool.BLL.Models.Vacations
{
    public class AnnualVacationFormModel : VacationFormModel
    {
        public DateTime StartOfWorkingYear { get; set; }
        public DateTime EndOfWorkingYear { get; set; }
        public int Days { get; set; }
        public IEnumerable<AnnualVacationModel> AnnualVacations { get; set; }
    }
}
